packages/wonder-blocks-button/src/components/button-core.tsx

```diff
@@ -331,170 +329,163 @@ export const _generateStyles = (
-                ":active": activePressedStyling,
             },
-            focused: focusStyling,
-            pressed: activePressedStyling,
-            disabled: {
-                ...disabledStateStyles,
-                cursor: "not-allowed",
-                // Reset hover/press styles when disabled
-                ":hover": sharedDisabledStyling,
-                ":active": sharedDisabledStyling,
-                ":focus-visible": focusStyling,
+
+            ":active": {
+                // shared
+                background: themeVariant.press.background,
+                color: themeVariant.press.foreground,
+                // primary
+                ...(kind === "primary"
+                    ? {
+                          outline: `${borderWidthKind.press} solid ${themeVariant.press.border}`,
+                          outlineOffset: outlineOffsetKind,
+                      }
+                    : undefined),
+                // secondary
+                ...(kind === "secondary"
+                    ? {
+                          borderColor: themeVariant.press.border,
+                          boxShadow: `inset 0 0 0 ${borderWidthKind.press} ${themeVariant.press.border}`,
+                      }
+                    : undefined),
+
+                // tertiary-specific styles
+                ...(kind === "tertiary"
+                    ? {
+                          textDecoration: "underline",
+                          textUnderlineOffset: theme.root.font.offset.default,
+                          textDecorationThickness:
+                              theme.root.sizing.underline.press,
+                      }
+                    : undefined),
             },
-        };
-    } else {
-        throw new Error("Button kind not recognized");
-    }
+
+            ...focusStyles.focus,
+            // These overrides are needed to ensure that the boxShadow is
+            // properly applied to the button when it is focused +
+            // hovered/pressed.
+            ...(kind === "secondary"
+                ? {
+                      ":focus-visible:hover": {
+                          ...focusStyles.focus[":focus-visible"],
+                          boxShadow: `inset 0 0 0 ${borderWidthKind.hover} ${themeVariant.hover.border}, ${focusStyles.focus[":focus-visible"].boxShadow}`,
+                      },
+                      ":focus-visible:active": {
+                          ...focusStyles.focus[":focus-visible"],
+                          boxShadow: `inset 0 0 0 ${borderWidthKind.press} ${themeVariant.press.border}, ${focusStyles.focus[":focus-visible"].boxShadow}`,
+                      },
+                  }
+                : {}),
+        },
+        disabled: {
+            cursor: "not-allowed",
+            ...disabledStatesStyles,
+            // NOTE: Even that browsers recommend to specify pseudo-classes in
+            // this order: link, visited, hover, focus, active, we need to
```

Comment:
Should the comment be in this order so it matches the implementation? 
```suggestion
            // this order: link, visited, hover, active, focus, we need to
```

Reply:
That comment is intended to present how browsers order these pseudo-classes. According to the article in the comments, that's how it would work (but I'm not 100% sure if that's true).